pilot/model/validation.go

```diff
@@ -234,22 +232,55 @@ func (t Labels) Validate() error {
 
 // ValidateFQDN checks a fully-qualified domain name
 func ValidateFQDN(fqdn string) error {
-	if len(fqdn) > 255 {
-		return fmt.Errorf("domain name %q too long (max 255)", fqdn)
+	return appendErrors(checkDNS1123Preconditions(fqdn), validateDNS1123Labels(fqdn))
+}
+
+func ValidateWildcardDomain(domain string) error {
+	if err := checkDNS1123Preconditions(domain); err != nil {
+		return err
 	}
-	if len(fqdn) == 0 {
+	// We only allow wildcards in the first label; split off the first label (parts[0]) from the rest of the host (parts[1])
+	parts := strings.SplitN(domain, ".", 2)
+	if !IsWildcardDNS1123Label(parts[0]) {
+		return fmt.Errorf("domain name %q invalid (label %q invalid)", domain, parts[0])
```

Comment:
what is Label here?

Reply:
I just copied the wording from the original `ValidateFQDN` method, which uses the error string: `"domain name %q invalid (label %q invalid)"`. Happy to update if we want to use different language, though AFAIK RFC 1123 uses the language "label" to describe a dot-delineated sub-piece of a domain.

Reply:
Ah I see. I wanted clarification thats all. Odd term to use in the RFC though..